### ISSUE TITLE
define/contract: undefined;

### ISSUE THREAD

**Issue by lockie** [lexi-lambda/racket-vector-struct#1]
Hey.
When I'm trying to use this library on Racket v7.0, I get
```
> (require vector-struct)
> (vecstruct point (x y))
; define/contract: undefined;
;  cannot reference an identifier before its definition
;   in module: top-level
;   internal name: define/contract
; [,bt for context]
```
`require`ing `racket/contract` does not help - the same error pops up. How can I fix this? Does the library need some updating to reflect changes in recent Racket versions? Please help.

**Comment by lexi-lambda** [comment]
This package is deprecated. Just use `struct`, instead! Is there a reason you want to use this package over structs?

**Comment by AlexKnauth** [comment]
I still use it, but only for Typed Racket projects.

**Comment by lockie** [comment]
@lexi-lambda Yeah, I'm working on a project with performance concerns, so I have to resort to some low-level optimizations like this one.

**Comment by lexi-lambda** [comment]
Have you benchmarked and found that using vectors is actually faster? If so, would you be willing to share the program with me? When I profiled structs versus vectors several versions back, I found that there wasn’t any difference in performance on newer versions of Racket.

**Comment by lockie** [comment]
As of now, I didn' make any benchmarks. I just thoght that using vector will give me the opportunity to set the whole struct values with one `memcpy` (quite large part of my project is written in pure C). As soon as I'll have numbers, I'll share them with you.
Right now I can't share the code, because it's a part of competition, I'll be able to do that in early february.

**Comment by AlexKnauth** [comment]
> I just thoght that using vector will give me the opportunity to set the whole struct values with one memcpy

I’m not sure whether Racket’s vectors work that way (although `fxvector`s and `flvector`s may, idk), and in untyped Racket, without type-based optimizations, I don’t think that’s likely at all.

**Comment by lockie** [comment]
@AlexKnauth It kinda works with the vector of integer objects, but right now I'm stuck with the fact that Racket sets the least significant bit of ints to 1 to indicate those are really Racket runtime objects.
